kubernetes/structures_stateful_set.go

```diff
@@ -232,10 +237,14 @@ func patchStatefulSetSpec(d *schema.ResourceData) (PatchOperations, error) {
 
 	if d.HasChange("spec.0.replicas") {
 		log.Printf("[TRACE] StatefulSet.Spec.Replicas has changes")
-		if v, ok := d.Get("spec.0.replicas").(int); ok {
+		if v, ok := d.Get("spec.0.replicas").(string); ok && v != "" {
+			vv, err := strconv.Atoi(v)
+			if err != nil {
+				return ops, err
+			}
 			ops = append(ops, &ReplaceOperation{
 				Path:  "/spec/replicas",
-				Value: v,
+				Value: vv,
```

Comment:
We don't have to add a `DeleteOperation` because when we set to `""` we want it to preserve whatever the current value of replicas is.